.goreleaser.yml

```diff
@@ -0,0 +1,60 @@
+archives:
+  - format: zip
+    name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+builds:
+  - env:
+      - CGO_ENABLED=0
+    flags:
+      - -trimpath
+    goos:
+      - darwin
+      - freebsd
+      - linux
+      - windows
+    goarch:
+      - '386'
+      - amd64
+      - arm
+      - arm64
+    ignore:
+      - goarch: arm
+        goos: windows
+      - goarch: arm64
+        goos: freebsd
+      - goarch: arm64
+        goos: windows
+    ldflags:
+      - -s -w -X internal/provider.Version={{.Version}}
+    mod_timestamp: '{{ .CommitTimestamp }}'
+checksum:
+  name_template: '{{ .ProjectName }}_{{ .Version }}_SHA256SUMS'
+  algorithm: sha256
+publishers:
+  - name: hc-releases
+    checksum: true
+    signature: true
+    cmd: hc-releases upload-file -header="x-terraform-protocol-version={{ .Env.PROTOCOL_VERSION }}" -header="x-terraform-protocol-versions={{ .Env.PROTOCOL_VERSIONS }}" {{ abs .ArtifactPath }}
+    env:
+      - AWS_DEFAULT_REGION={{ .Env.AWS_DEFAULT_REGION }}
+      - AWS_REGION={{ .Env.AWS_REGION }}
+      - AWS_ACCESS_KEY_ID={{ .Env.AWS_ACCESS_KEY_ID }}
+      - AWS_SECRET_ACCESS_KEY={{ .Env.AWS_SECRET_ACCESS_KEY }}
+      - AWS_SESSION_TOKEN={{ .Env.AWS_SESSION_TOKEN }}
+release:
+  ids:
+    - none
+signs:
```

Comment:
one thing that came up recently is that the artifacts found in https://releases.hashicorp.com for each provider directory should technically contain both this signature file with the public key in the filename and the `plain` filename as seen here for the google provider:

<img width="483" alt="Screen Shot 2021-11-18 at 12 03 10 AM" src="https://user-images.githubusercontent.com/13805595/142355128-675e1939-dd78-449a-9f8f-8f871f71bbc7.png">

so to do so with this new goreleaser configuration, you'll need to add an additional block that creates the "plain" signature file e.g.

```suggestion
signs:
  # Default Signature file (i.e. terraform-provider-awscc_VERSION_SHA256SUMS.sig)
  - cmd: sh
    args:
    - -c
    - >-
      signore
      sign
      --dearmor
      --file ${artifact}
      --signer {{ .Env.SIGNER }}
      --out ${signature}
    artifacts: checksum
  # Signature file with GPG Public Key ID in filename (i.e. terraform-provider-awscc_VERSION_SHA256SUMS.7685B676.sig)
  - id: sig-with-gpg-public-key-id
    signature: ${artifact}.72D7468F.sig
    cmd: sh
    args:
    - -c
    - >-
      signore
      sign
      --dearmor
      --file ${artifact}
      --signer {{ .Env.SIGNER }}
      --out ${signature}
    artifacts: checksum
```